libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2507,9 +2581,14 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 		case FunctionType::Kind::ABIEncodePacked:
 		case FunctionType::Kind::ABIEncodeWithSelector:
 		case FunctionType::Kind::ABIEncodeWithSignature:
+		case FunctionType::Kind::ABIEncodeCall:
 		{
 			typeCheckABIEncodeFunctions(_functionCall, functionType);
 			returnTypes = functionType->returnParameterTypes();
+
+			if (functionType->kind() == FunctionType::Kind::ABIEncodeCall)
+				typeCheckABIEncodeCallFunction(_functionCall);
```

Comment:
Wouldn't it be nicer to put this call inside `typeCheckABIEncodeFunctions`?

Reply:
True, it's kind of a subset of the other if you go by the naming